src/dotnet/APIView/APIViewWeb/Managers/CommentsManager.cs

```diff
@@ -119,32 +136,71 @@ public async Task<CommentModel> UpdateCommentAsync(ClaimsPrincipal user, string
             return comment;
         }
 
-        public async Task DeleteCommentAsync(ClaimsPrincipal user, string reviewId, string commentId)
+        /// <summary>
+        /// Delete Comment
+        /// </summary>
+        /// <param name="user"></param>
+        /// <param name="reviewId"></param>
+        /// <returns></returns>
+        public async Task SoftDeleteCommentsAsync(ClaimsPrincipal user, string reviewId)
+        {
+            var comments = await _commentsRepository.GetCommentsAsync(reviewId);
+
+            foreach (var  comment in comments)
+            {
+                await SoftDeleteCommentAsync(user, comment);
+            }
+        }
+
+        /// <summary>
+        /// Delete Comment
+        /// </summary>
+        /// <param name="user"></param>
+        /// <param name="reviewId"></param>
+        /// <param name="commentId"></param>
+        /// <returns></returns>
+        public async Task SoftDeleteCommentAsync(ClaimsPrincipal user, string reviewId, string commentId)
         {
             var comment = await _commentsRepository.GetCommentAsync(reviewId, commentId);
+            await SoftDeleteCommentAsync(user, comment);
+        }
+
+        /// <summary>
+        ///  Delete Comment
+        /// </summary>
+        /// <param name="user"></param>
+        /// <param name="comment"></param>
+        /// <returns></returns>
+        public async Task SoftDeleteCommentAsync(ClaimsPrincipal user, CommentItemModel comment)
+        {
             await AssertOwnerAsync(user, comment);
-            await _commentsRepository.DeleteCommentAsync(comment);
+            var changeUpdate = ChangeHistoryHelpers.UpdateBinaryChangeAction(comment.ChangeHistory, CommentChangeAction.Deleted, user.GetGitHubLogin());
```

Comment:
Is it possible to rename method `UpdateBinaryChangeAction` to something else to convey the use easily? Is it an async call?

Reply:
What's a better name? It's used to update the history when there are two possible states for that change e.g. `approved`, `approval reverted`, `open` `closed` that's why I called it a BinaryChange..